packages/react-native-editor/bin/generate-pot-files.sh

```diff
@@ -0,0 +1,243 @@
+
+  echo -e "\n\033[1mExtracting source files from \"$MAP_FILE\" source map file\033[0m"
+  node $SCRIPT_DIR/extract-files-from-sourcemap.js $MAP_FILE $TARGET_PATH
+}
+
+function generate_pot_files() {
+  local OUTPUT_PATH=$1
+  local PLUGIN_NAME=$2
+  local SOURCE_DIR=$3
+  shift 3
+  local PLUGINS_TO_SUBTRACT=( $@ )
+
+  # Define subtract pot files
+  local SUBTRACT_POT_FILES=()
+  for PLUGIN in "$@"; do
+    SUBTRACT_POT_FILES+=( "$OUTPUT_PATH/$PLUGIN-used-android.pot" )
+    SUBTRACT_POT_FILES+=( "$OUTPUT_PATH/$PLUGIN-used-ios.pot" )
+  done
+  local SUBTRACT_POT_FILES=$(join_by , ${SUBTRACT_POT_FILES[@]})
+
+  # Define output paths
+  local OUTPUT_POT_USED_ANDROID_FILE="$OUTPUT_PATH/$PLUGIN_NAME-used-android.pot"
+  local OUTPUT_POT_USED_IOS_FILE="$OUTPUT_PATH/$PLUGIN_NAME-used-ios.pot"
+  local OUTPUT_POT_BLOCKS_FILE="$OUTPUT_PATH/$PLUGIN_NAME-blocks.pot"
+  local OUTPUT_POT_SOURCE_FILE="$OUTPUT_PATH/$PLUGIN_NAME-source.pot"
+
+  local EXCLUDE_FILES="test/*,e2e-tests/*,bundle/*,build-module/*"
+
+  local DEBUG_PARAM=$([ -z $DEBUG ] && echo "" || echo "--debug")
+  local SUBTRACT_PARAM=$([ -z $SUBTRACT_POT_FILES ] && echo "" || echo "--subtract=$SUBTRACT_POT_FILES")
+  local DOMAIN_PARAM=$([ "$PLUGIN_NAME" == "gutenberg" ] && echo "--ignore-domain" || echo "--domain=$PLUGIN_NAME")
+
+  local MAKEPOT_COMMAND="$WP_CLI i18n make-pot"
+
+  echo -e "\n\033[1mExtract strings and generate POT files for \"$PLUGIN_NAME\" plugin from \"$SOURCE_DIR\"\033[0m"
+
+  mkdir -p $OUTPUT_PATH
+
+  if [ -n "$SUBTRACT_POT_FILES" ]; then
+    echo "--- Strings from ${PLUGINS_TO_SUBTRACT[@]} plugins will be subtracted ---"
+  fi
+  
+  echo -e "\nExtract used strings from Android source-map:"
+  $MAKEPOT_COMMAND $ANDROID_EXTRACT_SOURCE_FILES_PATH $DEBUG_PARAM $SUBTRACT_PARAM $DOMAIN_PARAM $OUTPUT_POT_USED_ANDROID_FILE
+
+  echo -e "\nExtract used strings from iOS source-map:"
+  $MAKEPOT_COMMAND $IOS_EXTRACT_SOURCE_FILES_PATH $DEBUG_PARAM $SUBTRACT_PARAM $DOMAIN_PARAM $OUTPUT_POT_USED_IOS_FILE
+
+  echo -e "\nExtract strings from block JSON files:"
+  $MAKEPOT_COMMAND $SOURCE_DIR $DEBUG_PARAM --exclude="$EXCLUDE_FILES" --skip-js --skip-php --ignore-domain $OUTPUT_POT_BLOCKS_FILE
+
+  echo -e "\nExtract strings from non-native JS code:"
+  $MAKEPOT_COMMAND $SOURCE_DIR $DEBUG_PARAM --exclude="$EXCLUDE_FILES" --merge="$OUTPUT_POT_BLOCKS_FILE" $DOMAIN_PARAM $OUTPUT_POT_SOURCE_FILE
+}
+
+# Get parameters
+PLUGINS=( "$@" )
+
+# Validate parameters
+if [[ $((${#PLUGINS[@]}%2)) -ne 0 ]]; then
+  error "Plugin arguments must be be even."
+fi
+
+for (( index=0; index<${#PLUGINS[@]}; index+=2 )); do
+  PLUGIN_FOLDER=${PLUGINS[index+1]}
+
+  if [[ ! -d $PLUGIN_FOLDER ]]; then
+    NOT_FOUND_PLUGIN_FOLDERS+=( $PLUGIN_FOLDER )
+    echo -e "\033[0;31mPlugin folder \"$PLUGIN_FOLDER\" doesn't exist.\033[0m"
+  fi
+done
+if [[ -n $NOT_FOUND_PLUGIN_FOLDERS ]]; then
+  exit 1
+fi
+
+# Define constants
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
```

Comment:
This variable is used to get the path of the script for calculating the path of other resources.